practice-app/website/api/copyright.py

```diff
@@ -0,0 +1,187 @@
+    if(not infringement_art_item):
+        return {"error": f"Art Item with id : {request_json['infringement_art_item_id']} does not exists."}, 404
+
+    # Checking if the given art items are different
+    if (request_json["original_art_item_id"] == request_json["infringement_art_item_id"]):
+        return {"error": "Original art item and infringement art item must be different."}, 400
+
+    # Getting a similarity score by calling an external API to compare two images
+    #   Smaller the value, similar the images
+    sim_score = get_similarity_score(original_art_item.content_uri, infringement_art_item.content_uri)
+
+    if (sim_score == -1):
+        return {"error": "Something went wrong in the external Image Similarity API. Please try again."}, 400
+
+    verify_jwt_in_request()
+
+    new_copyright_report = CopyrightInfringementReport(
+        creator = current_user.id,
+        original_art_item_id = request_json["original_art_item_id"],
+        infringement_art_item_id = request_json["infringement_art_item_id"],
+        description = request_json["description"],
+        similarity_score = sim_score,
+        creation_date = date.today()
+    )
+
+    # Creating the report and commiting it 
+    try:
+        db.session.add(new_copyright_report)
+        db.session.commit()
+    except Exception as error:
+        return {"error" : f"There was an error adding the copyright infringement report to the database, Error: {error}."}, 500
+
+    # Success
+    return {"id": new_copyright_report.id}, 201
+
+# API endpoint for viewing a copyright infringement report
+@copyright.route("/copyright/<report_id>", methods=["GET"])
+def get_copyright_report_data(report_id):
+
+    """
+    file: ./doc/copyright_GET.yml
+    """
+
+    report = CopyrightInfringementReport.query.get(report_id)
+
+    # Checking if report with given id exists
+    if not report:
+        return {"error": f"There are no reports with the id {report_id}."}, 404    
+
+    report = report.serialize()
+
+    # Giving the name of the reoprt creator to the frontend
+    report["creator_name"]  = User.query.get(report["creator"]).get_name()
+
+    # Giving relevant art items data to the frontend
+    original_art_item = ArtItem.query.get(report["original_art_item_id"])
+    infringement_art_item = ArtItem.query.get(report["infringement_art_item_id"])
+
+    report["original_art_name"]  = original_art_item.name
+    report["original_art_creation_date"]  = original_art_item.creation_date
+    report["original_art_uri"]  = original_art_item.content_uri
+    report["infringement_art_name"]  = infringement_art_item.name
+    report["infringement_art_creation_date"]  = infringement_art_item.creation_date
+    report["infringement_art_uri"]  = infringement_art_item.content_uri
+
+    return report, 200
+
+
+# API endpoint for removing an art item
+@copyright.route("/copyright", methods=["DELETE"])
+def remove_art_item():
```

Comment:
This should require an artist token (or user token depending on your implementation)

Reply:
It actually should require a admin user, since we did not implement an admin user type, i just left it as accessible by any user

Reply:
requested changes are done [relevant commit](https://github.com/bounswe/bounswe2022group7/pull/201/commits/a5b3c232f1bf254240b0f4e9f821c13bda45c3ea)